src/basic/smack-util.h

```diff
@@ -29,8 +29,21 @@ bool mac_smack_use(void);
 
 int mac_smack_fix(const char *path, bool ignore_enoent, bool ignore_erofs);
 
+int mac_smack_read(const char *path, const char **label);
 int mac_smack_apply(const char *path, const char *label);
+int mac_smack_read_fd(int fd, const char **label);
 int mac_smack_apply_fd(int fd, const char *label);
-int mac_smack_apply_pid(pid_t pid, const char *label);
-int mac_smack_apply_ip_in_fd(int fd, const char *label);
+int mac_smack_read_exec(const char *path, const char **label);
+int mac_smack_apply_exec(const char *path, const char *label);
+int mac_smack_read_exec_fd(int fd, const char **label);
+int mac_smack_apply_exec_fd(int fd, const char *label);
+int mac_smack_read_ip_out(const char *path, const char **label);
+int mac_smack_apply_ip_out(const char *path, const char *label);
+int mac_smack_read_ip_out_fd(int fd, const char **label);
 int mac_smack_apply_ip_out_fd(int fd, const char *label);
+int mac_smack_read_ip_in(const char *path, const char **label);
+int mac_smack_apply_ip_in(const char *path, const char *label);
+int mac_smack_read_ip_in_fd(int fd, const char **label);
+int mac_smack_apply_ip_in_fd(int fd, const char *label);
+
+int mac_smack_apply_pid(pid_t pid, const char *label);
```

Comment:
Hmm, I wonder whether we can simplify that list of functions with an enum, something like this:

```c
enum {
    SMACK_LABEL_EXEC,
    SMACK_LABEL_IP_IN,
    SMACK_LABEL_IP_OUT,
    ....
};
```
and then make `mac_smack_{read,apply}{_fd,}()` generic by taking a value of this enum. That would reduce the exposed interface to exactly 4 functions.

@poettering, any opinion?